src/workers/squadFeatureAccess.ts

```diff
@@ -0,0 +1,27 @@
+import { messageToJson, Worker } from './worker';
+import { ChangeObject } from '../types';
+import { SourceMember, Feature, FeatureType } from '../entity';
+
+interface Data {
+  sourceMember: ChangeObject<SourceMember>;
+}
+
+const worker: Worker = {
+  subscription: 'api.squad-feature-access',
+  handler: async (message, con): Promise<void> => {
+    const { sourceMember: member }: Data = messageToJson(message);
+    const variables = {
+      feature: FeatureType.Squad,
+      userId: member.userId,
+    };
+    const hasAccess = await con.getRepository(Feature).findOneBy(variables);
+
+    if (hasAccess) {
+      return;
+    }
+
+    await con.getRepository(Feature).insert(variables);
```

Comment:
I'd like to suggest doing only the insert with try and catch. If the error thrown is due to duplicate we just ignore it otherwise we throw the error. Will turn 2 queries into one :)

Reply:
Ah good point! 🙏